src/eventmaster/api_handlers.go

```diff
@@ -382,30 +382,24 @@ func (h *httpHandler) handleGetDc(w http.ResponseWriter, r *http.Request, _ http
 	h.sendResp(w, "", string(str), r.URL.Path)
 }
 
-func (h *httpHandler) handleGitHubEvent(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	var info map[string]interface{}
+func (h *httpHandler) handlePluginEvent(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	pluginName := ps.ByName("name")
+	if plugin, ok := Plugins[pluginName]; ok {
 
-	reqBody, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error reading request body", r.URL.Path)
-		return
-	}
-	if err := json.Unmarshal(reqBody, &info); err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error JSON decoding body of request", r.URL.Path)
-		return
-	}
-
-	id, err := h.store.AddEvent(&UnaddedEvent{
-		Dc:        "github",
-		Host:      "github",
-		TopicName: "github",
-		Data:      info,
-	})
-	if err != nil {
-		h.sendError(w, http.StatusInternalServerError, err, "Error adding event to store", r.URL.Path)
-		return
+		event, err := plugin.ParseRequest(r)
```

Comment:
If we are going to have the relatively strict "parse" and then "add" events, we can probably make these plugins a bit layered to reduce the amount of code required.

Right now this is a single pass `payload -> event` -- if we instead change it to `payload -> data -> event` we can separate out the payload parsing from the event creation. If we do that we'd need a similar layer for "auth". The idea here being that a user could create a new "plugin" handler without writing code. Probably worth looking into